src/java/main/org/apache/zookeeper/server/NIOServerCnxn.java

```diff
@@ -374,14 +373,12 @@ void doIO(SelectionKey k) throws InterruptedException {
             // expecting close to log session closure
             close();
         } catch (EndOfStreamException e) {
-            LOG.warn("caught end of stream exception",e); // tell user why
-
+            LOG.warn(e.getMessage());
```

Comment:
Do we want to log the whole stack trace if debug is enabled here?

Reply:
Yes, it would make sense to log the stack trace at debug level for EndOfStreamException too, so that we don't get less information in the log after this change is applied. The reason why I did not include it first, was to keep this change a pure backport from master and branch-3.5 to branch-3.4 as the original change does not include this additional logging statement. I've added it now, but probably then we should also add it to master as well if we wish to keep it.